scripts/build/builders/realtek.py

```diff
@@ -0,0 +1,151 @@
+            return 'chip-rtl8777g-lock-app'
+        elif self == RealtekApp.WINDOW:
+            return 'chip-rtl8777g-window-app'
+        else:
+            raise Exception('Unknown app type: %r' % self)
+
+
+class RealtekBuilder(Builder):
+
+    def __init__(self,
+                 root,
+                 runner,
+                 board: RealtekBoard = RealtekBoard.RTL8777G,
+                 app: RealtekApp = RealtekApp.LIGHT,
+                 enable_cli: bool = False,
+                 enable_rpc: bool = False,
+                 enable_shell: bool = False):
+        super(RealtekBuilder, self).__init__(root, runner)
+        self.board = board
+        self.app = app
+        self.enable_cli = enable_cli
+        self.enable_rpc = enable_rpc
+        self.enable_shell = enable_shell
+
+        self.ot_src_dir = os.path.join(os.getcwd(), 'third_party/openthread/ot-realtek')
+
+    def CmakeBuildFlags(self):
+        flags = []
+
+        flags.append("-DCMAKE_BUILD_TYPE=Release")
+        flags.append("-DCMAKE_TOOLCHAIN_FILE=src/bee4/arm-none-eabi.cmake")
+        flags.append("-DBUILD_TYPE=sdk")
+        flags.append(f"-DBUILD_TARGET={self.board.BoardName}")
+        flags.append(f"-DBUILD_BOARD_TARGET={self.board.BoardName}")
+        flags.append(f"-DOT_CMAKE_NINJA_TARGET={self.app.TargetName}")
+        flags.append(f"-DMATTER_EXAMPLE_PATH={self.root}/examples/{self.app.ExampleName}/realtek_bee")
+
+        if self.enable_cli:
+            flags.append("-DENABLE_CLI=ON")
+        else:
+            flags.append("-DENABLE_CLI=OFF")
+
+        if self.enable_rpc:
+            flags.append("-DENABLE_PW_RPC=ON")
+        else:
+            flags.append("-DENABLE_PW_RPC=OFF")
+
+        if self.enable_shell:
+            flags.append("-DENABLE_SHELL=ON")
+        else:
+            flags.append("-DENABLE_SHELL=OFF")
+
+        build_flags = " ".join(flags)
+
+        return build_flags
+
+    def generate(self):
+        cmd = 'arm-none-eabi-gcc -D BUILD_BANK=0 -E -P -x c {ot_src_dir}/src/bee4/{board_name}/app.ld -o {ot_src_dir}/src/bee4/{board_name}/app.ld.gen'.format(
+            ot_src_dir=self.ot_src_dir,
+            board_name=self.board.BoardName)
+        self._Execute(['bash', '-c', cmd])
+
+        cmd = 'cmake -GNinja -DOT_COMPILE_WARNING_AS_ERROR=ON {build_flags} {example_folder} -B{out_folder}'.format(
+            build_flags=self.CmakeBuildFlags(),
+            example_folder=self.ot_src_dir,
+            out_folder=self.output_dir)
+
+        self._Execute(['bash', '-c', cmd], title='Generating ' + self.identifier)
+
+    def _build(self):
+        cmd = ['ninja', '-C', self.output_dir]
+
+        if self.ninja_jobs is not None:
+            cmd.append('-j' + str(self.ninja_jobs))
+
+        cmd.append(self.app.TargetName)
+
+        self._Execute(cmd, title='Building ' + self.identifier)
+
+        os.system(f"rm -rf {self.root}/third_party/openthread/ot-realtek/src/bee4/{self.board.BoardName}/*.gen")
```

Comment:
please use self._Execute always to allow for `--dry-run` to work. We intentionally do not use python API or direct execution.